src/encode.rs

```diff
@@ -45,37 +45,47 @@ pub fn write_zlib_compound_tag<W: Write>(
 /// root_tag.insert_compound_tag_vec("servers", servers);
 ///
 /// let mut vec = Vec::new();
-/// write_compound_tag(&mut vec, root_tag).unwrap();
+/// write_compound_tag(&mut vec, &root_tag).unwrap();
 /// ```
 pub fn write_compound_tag<W: Write>(
     writer: &mut W,
-    compound_tag: CompoundTag,
+    compound_tag: &CompoundTag,
 ) -> Result<(), Error> {
-    let name = compound_tag.name.clone();
-    let tag = Tag::Compound(compound_tag);
-    writer.write_u8(tag.type_id())?;
+    // Tag id
+    writer.write_u8(10)?;
```

Comment:
Why not keep `writer.write_u8(tag.type_id())?;`?

Reply:
It does, yeah. I wrote it because type_id() is an instance method and we don't have an instance (and the constant will never change), but the right thing to use would probably be `Tag::Compound(CompoundTag::new()).type_id()`.